src/kayenta/report/detail/detail.tsx

```diff
@@ -1,39 +1,48 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
 
+import { ICanaryExecutionStatusResult } from 'kayenta/domain';
+import { ICanaryState } from 'kayenta/reducers';
 import ReportHeader from './header';
 import ReportScores from './reportScores';
 import MetricResults from './metricResults';
+import ReportExplanation from './reportExplanation';
+import ReportException from './reportException';
 
 import './detail.less';
-import ReportExplanation from './reportExplanation';
 
 /*
  * Layout for report detail view.
  * */
 
-export interface IDetailViewState {
-  isExpanded: boolean;
+interface IDetailViewProps {
+  run: ICanaryExecutionStatusResult;
 }
 
-export default class DetailView extends React.Component<any, IDetailViewState> {
-  public state: IDetailViewState = { isExpanded: true };
-
-  private toggleDetailHeader = (): void => {
-    this.setState({ isExpanded: !this.state.isExpanded });
-  };
+function DetailView({ run }: IDetailViewProps) {
+  const [isExpanded, setIsExpanded] = React.useState<boolean>(true);
 
-  public render() {
-    const { isExpanded } = this.state;
-
-    return (
-      <>
-        <div className="vertical flex-1">
-          {isExpanded && <ReportHeader />}
-          <ReportExplanation />
-          <ReportScores isExpanded={isExpanded} toggleHeader={this.toggleDetailHeader} />
-          <MetricResults />
-        </div>
-      </>
-    );
+  if (!run.result?.judgeResult.results?.length && run.exception) {
```

Comment:
I believe the exception object can be there and empty. In the backend, I had to check that there was an exceptionType present on the object as well. Something to check on.

Reply:
In the exception case, won't judgeResult be undefined which may cause this to explode (I can't remember exactly how the ? operator works.)